crates/loro-internal/src/encoding/encode_reordered.rs

```diff
@@ -481,17 +472,28 @@ pub(crate) fn encode_snapshot(oplog: &OpLog, state: &DocState, vv: &VersionVecto
 
     let mut states = Vec::new();
     let mut state_bytes = Vec::new();
-    let peer_register = Rc::new(RefCell::new(peer_register));
-    let peer_register_1 = Rc::clone(&peer_register);
+    for (_, container) in c_pairs.iter() {
+        let container_index = *container_idx2index.get(container).unwrap() as u32;
+
+        let is_unknown = container.is_unknown();
+
+        if is_unknown {
```

Comment:
Need comments here. And I think the code structure would be better if we extract the encoder and decoder of the container state